parsec/include/parsec/sys/atomic-macosx.h

```diff
@@ -44,17 +39,6 @@ int parsec_atomic_cas_int64( volatile int64_t* location,
     return OSAtomicCompareAndSwap64( old_value, new_value, location );
 }
 
-#if defined(PARSEC_ATOMIC_USE_GCC_128_BUILTINS)
```

Comment:
clang/llvm/gcc/icc on OSX all provide __sync_bool_compare_and_swap, and this intrinsic can be used in addition to the default OSX atomic support. Instead of removing it we should make sure it is in use when available.

Reply:
I hesitated, but we didn't do this before, on Apple we would force disable the int128 type altogether, and this code would never actually be used. 

If we are using the GCC builtins for the 128 bit, why not use them all around and just drop the OSX  specific OSAtomics? (that is, drop down to the atomic-gcc.h). Is there a convincing performance advantage for maintaining this code?

Reply:
That makes more sense. When atomic support was nascent we needed to deal with what was available, but today, as atomic support become mainstream, we don't necessarily need all the special cases.

Let's remove the OSX specific atomic support.